NV2ALog.py

```diff
@@ -0,0 +1,51 @@
+"""Manages the nv2a log file."""
+
+# pylint: disable=invalid-name
+# pylint: disable=consider-using-f-string
+
+Nv2aLogMethodDetails = False
+
+
+class NV2ALog:
+    """Manages the nv2a log file."""
+
+    def __init__(self, path):
+        self.path = path
+
+        with open(self.path, "w", encoding="utf8") as logfile:
+            logfile.write("xemu style NV2A log from nv2a-trace.py\n\n")
```

Comment:
I still don't like xemu and don't agree that it should set standards for how tools should operate / format their output. At the very least, I'd rather not have it mentioned in this codebase - this tool is made for hardware, not for emulators like xemu.

In general, I'd suggest to split the display of the captured buffers into a separate step.

Reply:
> I'm not sure I follow; do you mean store all the info currently written into the HTML and nv2a logs and then process into separate outputs?

One could just dump the raw pushbuffers in the capturing phase, doing minimal processing on them.
Then, in a second phase, decode these into HTML or other log formats.

I expect future improvements to the decoding phase, whereas the capturing phase likely won't change much.

During development of nv2a-trace, I had collected numerous traces of interesting scenes, but often they became outdated because of missing data in captures or the decoded portion.
By capturing more data and decoding later, I was able to improve my decoders without having to recapture a large amount of scenes.
So I think it's best to decouple log generation from the capturing / have a log format which stores as much information as possible (even if it might not be human readable).

Reply:
Totally makes sense to me, particularly since it would not be very hard to run the decoder in a separate thread to retain the current sense of progress. I'd be happy to take a stab at this as a followup.